File: ClientApp/src/app/clases/evento.ts

export class evento
{
    id: number;
    nombre: string;
    fechainicio: Date;
    fechafin: Date;
    tipo: string;
    idmateria:number;

    constructor(obj:any)
    {
        this.id = +obj.id;
        this.nombre = obj.nombre;
        this.fechainicio = new Date(obj.fechainicio);
        this.fechafin = new Date(obj.fechafin);
        this.tipo = obj.tipo;
        this.idmateria = +obj.idmateria;
    }
}

File: modelo/evento.cs
using System.Data;
using System;
public class evento : oObjeto
{
    public string Nombre { get; set; }
    public DateTime Fechainicio{ get; set; }
    public DateTime Fechafin { get; set; }
    public string Tipo {get;set;}
    public int Idmateria{get;set;} 
    public evento() { }
    public evento(DataRow dr)
    {
        this.Id = Convert.ToInt32(dr["id"]);
        this.Nombre = dr["nombre"].ToString();
        this.Fechainicio = Convert.ToDateTime(dr["fechainicio"]);
        this.Fechafin= Convert.ToDateTime(dr["fechafin"]);
        this.Tipo= dr["tipo"].ToString();
        this.Idmateria=Convert.ToInt32(dr["idmateria"]);
    }

}